services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java

```diff
@@ -497,9 +498,20 @@ public Response authResponse(@QueryParam(AbstractOAuth2IdentityProvider.OAUTH2_P
                 }
 
                 if (authorizationCode != null) {
-                    String response = generateTokenRequest(authorizationCode).asString();
-
-                    BrokeredIdentityContext federatedIdentity = getFederatedIdentity(response);
+                    final SimpleHttp request = generateTokenRequest(authorizationCode);
+                    String response = null;
+                    BrokeredIdentityContext federatedIdentity;
+                    try {
+                        response = request.asString();
+                        federatedIdentity = getFederatedIdentity(response);
+                    } catch (Exception ex) {
+                        logger.debug(String.format("Token request failed to yield an authenticated identity. Request{%s %s [%s]} -> Response{%s}",
+                            request.method(), request.url(),
```

Comment:
What about to obtain some information by `getConfig()` method? For _url_ it could be `getConfig().getTokenUrl()`,... WDYT?

Reply:
I was thinking that it was more robust to use the actual url and method of the actual request. Currently `getConfig().getTokenUrl()` would return what all subclasses are currently using, but there is no guarantee that any overrides of generateTokenRequest will actually use this url, so using `getConfig().getTokenUrl()` for the debug logging would potentially log incorrect information.